core/src/node_interface/node_api.rs

```diff
@@ -95,6 +96,25 @@ impl NodeApi {
         );
         Ok(self.node.submit_transaction(&signed_tx)?)
     }
+
+    /// Unlock wallet
+    pub fn wallet_unlock(&self, password: &str) -> Result<bool, NodeApiError> {
```

Comment:
Could we use the node method? `self.node.wallet_unlock`?

Reply:
Unfortunately not since oracle-core is relying on an unmerged branch for `ergo-node-interface-rust`:
```toml
ergo-node-interface = { git = "https://github.com/ergoplatform/ergo-node-interface-rust", rev = "143c2a3dc8fb772d1af37f1f1e1924067c6aad14" }
```
and the recently released version is missing those functions. Also the node-interface is relying on a newer version of `ergo-lib` which would require a larger refactoring of `oracle-core` I am currently not comfortable doing.